src/amber/router/route.cr

```diff
@@ -1,5 +1,5 @@
 module Amber
-  class Route
+  struct Route
```

Comment:
This seems like a big change, and I'm not sure it directly relates to the HEAD requests, does it?

Reply:
This was an improvement that I've been wanting to make. This has to do more with the inner workings of crystal. In short, the Route is just holding a struct of data.

For a better understanding of this change read this https://crystal-lang.org/docs/guides/performance.html

Reply:
I've read about the performance gains before, a struct goes on the stack and is immutable (or copied by value anyway).

So I think it will be an improvement, but I ran into issues when working on a caching solution where I tried a struct, but it didn't work because I needed parts to be mutable.

I suppose a route should never change during run time - it just seems like a non-zero risk, and putting it in here means if we rollback, then we rollback the HEAD functionality too.